dpbench/benchmarks/black_scholes/black_scholes_numba_mlir_n.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache 2.0
+
+
+from math import erf
+
+import numba_mlir as nb
+from numpy import exp, log, sqrt
+
+
+@nb.vectorize(nopython=True)
+def _nberf(x):
+    return erf(x)
+
+
+@nb.njit(parallel=True, fastmath=True)
+def _black_scholes(price, strike, t, rate, volatility, call, put):
+    mr = -rate
+    sig_sig_two = volatility * volatility * 2
+
+    P = price
+    S = strike
+    T = t
+
+    a = log(P / S)
+    b = T * mr
+
+    z = T * sig_sig_two
+    c = 0.25 * z
+    y = 1.0 / sqrt(z)
+
+    w1 = (a - b + c) * y
+    w2 = (a - b - c) * y
+
+    d1 = 0.5 + 0.5 * _nberf(w1)
+    d2 = 0.5 + 0.5 * _nberf(w2)
+
+    Se = exp(b) * S
+
+    r = P * d1 - Se * d2
+    call[:] = r  # temporary `r` is necessary for faster `put` computation
+    put[:] = r - P + Se
+
+
+def black_scholes(nopt, price, strike, t, rate, volatility, call, put):
+    _black_scholes(price, strike, t, rate, volatility, call, put)
```

Comment:
This function is not needed. We can rename `_black_scholes` as `black_scholes` and have the framework call the jit-ed function directly.

Reply:
I'm getting weird errors without this wrapper, I'll leave it for separate investigation